### PR TITLE
RFC: Add View::get_mut_n as a safe alterantive to View::get_mut_unchecked.

### LINKED ISSUE #217: get_mut2/query_one2 (or similar functionality?)
There are a few cases where it would be very useful to be able to reference multiple distinct entities w/ potentially the same components, at the same time. The only way this is currently possible is through the `get_unchecked` methods, or with a query which is later filtered out or collected into an auxiliary structure. Unless I've missed something?

### PR DESCRIPTION
**Pull request by adamreichold**
Fixes #217

### CODE REVIEW

**Comment by adamreichold** [comment]
`PreparedView::get_mut_n` is missing as I wanted to fix the design first.

**Comment by Ralith** [comment]
Design LGTM.

**Comment by adamreichold** [comment]
Pushed the version including `PreparedView::get_mut_n` and a test case trying to get aliasing mutable references. I also changed the self type to `&mut self` as we still need exclusive access to the view even if the entities are distinct (as we could just call the method twice to produce overlapping exclusive references otherwise).

File: src/query.rs [hunk 1]
```diff
@@ -1394,8 +1433,46 @@ impl<'q, Q: Query> PreparedView<'q, Q> {
             .as_ref()
             .map(|fetch| fetch.get(meta.location.index as usize))
     }
+
+    /// Like `get_mut`, but allows checked simultaneous access to multiple entities
+    ///
+    /// For N > 3, the checking will clone the array and take O(N log N) time.
```
Comment (Ralith) [inline src/query.rs:1439]: If we're going to make one of multiple similar methods the canonical one and give it more detailed docs, we should link to it from the non-canonical methods (and probably omit the duplicated details entirely).

**Review by Ralith — CHANGES_REQUESTED** [review]
> I also changed the self type to &mut self

Whoops, good catch!

File: src/query.rs (hunk 1 above)
Comment (adamreichold) [reply src/query.rs]: Fixed as suggested.

**Comment by Ralith** [comment]
Thanks!

**Pull request merged**